src/hooks/useElementVisible.js

```diff
@@ -0,0 +1,21 @@
+import { useLayoutEffect, useState } from 'preact/hooks';
+
+export function useElementVisible(element) {
+
+  const [ visible, setVisible ] = useState(false);
+
+  useLayoutEffect(() => {
+    if (!element) return;
+
+    const resizeObserver = new ResizeObserver(([ entry ]) => {
+
+      // Prevents ResizeObserver from going into infinite loop.
+      requestAnimationFrame(() => setVisible(!!entry.contentRect.height));
```

Comment:
If I understand it correctly, the resize observer may still request multiple animation frames. Perhaps we could cache the last known value and change `visible` value only on change?